### PR TITLE
TEST: added different dataframes testing in `sklearnex` own tests

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
Added different dataframes testing in `sklearnex` own tests for checking `dpnp` and `dpctl` interop in `sklearnex`.
* `_dataframes_support` modul in onedal tests utils. Will reuse for onedal4py tests also.
* Updated tests of modules:
   * `sklearnex.neignbors`
   * `sklearnex.preview.decomposition`
   * `sklearnex.preview.ensemble`
   * `sklearnex.preview.linear_model`
   * `sklearnex.svm`

Before:
```python
neighbors/tests/test_neighbors.py::test_sklearnex_import_knn_classifier PASSED [  3%]
```

After:
```python
neighbors/tests/test_neighbors.py::test_sklearnex_import_knn_classifier[numpy] PASSED [  3%]
neighbors/tests/test_neighbors.py::test_sklearnex_import_knn_classifier[dpctl-SyclQueue_CPU] PASSED [  4%]
neighbors/tests/test_neighbors.py::test_sklearnex_import_knn_classifier[dpctl-SyclQueue_GPU] PASSED [  4%]
neighbors/tests/test_neighbors.py::test_sklearnex_import_knn_classifier[dpnp-SyclQueue_CPU] PASSED [  4%]
neighbors/tests/test_neighbors.py::test_sklearnex_import_knn_classifier[dpnp-SyclQueue_GPU] PASSED [  4%]
```

# TODO:
* Update testing for sklearnex and enable `deselcted_tests.yaml` for sklearnex tests. -> in seperate PR:
  * remove TODOs from sklearnex tests and add test cases into deselect_tests.yaml list
     *  see: https://github.com/intel/scikit-learn-intelex/pull/1379#issuecomment-1659340064
* Wide description of the PR :)
* revert https://github.com/intel/scikit-learn-intelex/pull/1379/commits/960071d7cbed1093c225d66a17734fb359a0e04e

### CODE REVIEW

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee2f8c8c-fcab-f15e-8d3e-a4bf010d0e2e. Specifically for sklearnex examples GPU.

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee2fad8c-fbe5-f146-abb5-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU without `dpnp` on env.

**Comment by samir-nasibli** [comment]
In this PR, the number of sklearnex own test cases has increased. Added `dpnp` and `dpctl` dataframes with cpu and gpu contexts to certain tests. So we have new 56 test cases, which checks dpnp/dpcttl interop.

Unfortunately, there are 18 new failures for different reasons, that should be addressed in separate PRs.

```python
=========================== short test summary info ============================
FAILED neighbors/tests/test_neighbors.py::test_sklearnex_import_nn[dpctl-SyclQueue_CPU]
FAILED neighbors/tests/test_neighbors.py::test_sklearnex_import_nn[dpctl-SyclQueue_GPU]
FAILED neighbors/tests/test_neighbors.py::test_sklearnex_import_nn[dpnp-SyclQueue_CPU]
FAILED neighbors/tests/test_neighbors.py::test_sklearnex_import_nn[dpnp-SyclQueue_GPU]
FAILED preview/ensemble/tests/test_preview_ensemble.py::test_sklearnex_import_rf_regression[dpctl-SyclQueue_GPU]
FAILED preview/ensemble/tests/test_preview_ensemble.py::test_sklearnex_import_rf_regression[dpnp-SyclQueue_GPU]
FAILED preview/ensemble/tests/test_preview_ensemble.py::test_sklearnex_import_et_classifier[dpctl-SyclQueue_GPU]
FAILED preview/ensemble/tests/test_preview_ensemble.py::test_sklearnex_import_et_classifier[dpnp-SyclQueue_GPU]
FAILED preview/ensemble/tests/test_preview_ensemble.py::test_sklearnex_import_et_regression[dpctl-SyclQueue_GPU]
FAILED preview/ensemble/tests/test_preview_ensemble.py::test_sklearnex_import_et_regression[dpnp-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_nusvc[dpctl-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_nusvc[dpnp-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_svc[dpctl-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_svc[dpnp-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_svr[dpctl-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_svr[dpnp-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_nusvr[dpctl-SyclQueue_GPU]
FAILED svm/tests/test_svm.py::test_sklearnex_import_nusvr[dpnp-SyclQueue_GPU]
===== 18 failed, 363 passed, 7 skipped, 429 warnings in 157.47s (0:02:37) ======
```

Currently, I have disabled this test cases via test filters + `TODO` comment, like we do for the tests for onedal module.
For example
```python
# TODO:
# investigate failure for `dpnp.ndarrays` and `dpctl.tensors` on `GPU`
@pytest.mark.parametrize(
    "dataframe,queue", _get_dataframes_and_queues(device_filter_="cpu")
)
def test_sklearnex_import_et_classifier(dataframe, queue):
```

We don't have list of sklearnex or onedal deselected test like we have for sklearn tests, see [deselected_tests.yaml](https://github.com/intel/scikit-learn-intelex/blob/master/deselected_tests.yaml).

I think we should extend sklearnex testing and add support of deselected tests for onedal4py and sklearnex modules instead of using filters. This is new feature, that should be implemented in separate branch. See TODO in [description] (https://github.com/intel/scikit-learn-intelex/pull/1379#issue-1824437850)

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee2ffcc8-9bec-f1c1-a4e7-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU without dpnp on env.

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee2ffd93-f8c6-f1de-bd22-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU with `dpctl` and `dpnp`.

File: onedal/tests/utils/_dataframes_support.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    dataframes_and_queues = [
+        pytest.param("numpy", None, id="numpy"),
+    ]
+
+    def _get_df_and_q(dataframe: str):
+        for queue in get_queues(device_filter_):
+            id = "{}-SyclQueue_{}".format(
+                dataframe, "GPU" if queue.sycl_device.is_gpu else "CPU"
+            )
+            dataframes_and_queues.append(pytest.param(dataframe, queue, id=id))
```
Comment (Alexsandruss) [inline onedal/tests/utils/_dataframes_support.py:51]: Bad practice in this specific case: it's better to return list of new data frames and queues than update of existing `dataframes_and_queues` var. from higher scope

Comment (samir-nasibli) [reply onedal/tests/utils/_dataframes_support.py:51]: this is inner function, that uses or updates function variable. It doesn't makes for them, but I can update it with return values

Comment (samir-nasibli) [reply onedal/tests/utils/_dataframes_support.py]: done

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee318ac9-4082-f1b8-a4e7-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU with dpctl and dpnp.

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee318ac9-4082-f1b8-a4e7-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU with dpctl and dpnp.

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee321570-cbcb-f141-9c8d-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU with dpctl and dpnp.
Using onedal with [ee321570-cbcb-f141-9c8d-a4bf010d0e2e](https://github.com/oneapi-src/oneDAL/commit/ab7f43f15fb816ab61bf77157d6f165ef81532cf)

**Comment by samir-nasibli** [comment]
fix for linter check is on #1364

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee35e825-3d41-f126-b1a7-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU with dpctl and dpnp.

**Comment by samir-nasibli** [comment]
Job: http://intel-ci.intel.com/ee369d0a-5daa-f1ea-b9eb-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU with dpctl and dpnp.

**Comment by samir-nasibli** [comment]
Latest CI check: http://intel-ci.intel.com/ee382f7b-9a45-f121-94b4-a4bf010d0e2e

File: onedal/tests/utils/_dataframes_support.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    elif target_df == "numpy":
+        return np.asarray(obj, *args, **kwargs)
+    # DPNP ndarray.
+    elif target_df == "dpnp":
+        return dpnp.asarray(
+            obj, usm_type="device", sycl_queue=sycl_queue, *args, **kwargs
+        )
+    # DPCtl tensor.
+    elif target_df == "dpctl":
+        return dpt.asarray(obj, usm_type="device", sycl_queue=sycl_queue, *args, **kwargs)
```
Comment (KulikovNikita) [inline onedal/tests/utils/_dataframes_support.py:82]: What about the last `else` option? Shouldn't you raise an error?

**Review by KulikovNikita — APPROVED** [review]

File: onedal/tests/utils/_dataframes_support.py (hunk 2 above)
Comment (samir-nasibli) [reply onedal/tests/utils/_dataframes_support.py:82]: Good catch actually, let me update a little bit a logic for this function.

**Pull request merged**